ios-base/Networking/Services/AuthenticationServices.swift

```diff
@@ -9,168 +9,190 @@
 import Foundation
 import UIKit
 import RSSwiftNetworking
+import RSSwiftNetworkingAlamofire
 
 internal class AuthenticationServices {
-
-  enum AuthError: Error {
+  
+  enum AuthError: LocalizedError {
+    case login
+    case signUp
+    case logout
+    case mapping
+    case request
     case userSessionInvalid
+    
+    var errorDescription: String? {
+      switch self {
+      case .login:
+        return "authError_login".localized
+      case .signUp:
+        return "authError_signUp".localized
+      case .logout:
+        return "authError_logout".localized
+      case .mapping:
+        return "authError_mapping".localized
+      case .request:
+        return "authError_request".localized
+      case .userSessionInvalid:
+        return "authError_request".localized
+      }
+    }
   }
-
+  
   // MARK: - Properties
   
   fileprivate static let usersUrl = "/users/"
   fileprivate static let currentUserUrl = "/user/"
-
+  
   private let sessionManager: SessionManager
-
-  private let apiClient: APIClient
-
+  private let userDataManager: UserDataManager
+  private let apiClient = BaseAPIClient.alamofire
+  
   init(
     sessionManager: SessionManager = .shared,
-    apiClient: APIClient = iOSBaseAPIClient.shared
+    userDataManager: UserDataManager = .shared
   ) {
     self.sessionManager = sessionManager
-    self.apiClient = apiClient
+    self.userDataManager = userDataManager
   }
   
-  func login(
+  @discardableResult func login(
     email: String,
-    password: String,
-    completion: @escaping (Result<Void, Error>) -> Void
-  ) {
-    apiClient.request(
+    password: String
+  ) async -> Result<UserData, AuthError> {
+    let response: RequestResponse<UserData> = await apiClient.request(
       endpoint: AuthEndpoint.signIn(email: email, password: password)
-    ) { [weak self] (result: Result<User?, Error>, responseHeaders: [AnyHashable: Any]) in
-      switch result {
-      case .success(let user):
-        if self?.saveUserSession(user, headers: responseHeaders) ?? false {
-          completion(.success(()))
-        } else {
-          completion(.failure(AuthError.userSessionInvalid))
-        }
-      case .failure(let error):
-        completion(.failure(error))
+    )
+    switch response.result {
+    case .success(let user):
+      if
+        let user = user,
```

Comment:
FYI This is possible to do now  in swift 5.7 xD 

```suggestion
        let user
```